Add generate migration to the documentation (#1429)
strapi/documentation: docs/developer-docs/latest/developer-resources/database-migrations.md

Database migrations are experimental. This feature is still a work in progress and will continue to be updated and improved. In the meantime, feel free to ask for help on the [forum](https://forum.strapi.io/) or on the community [Discord](https://discord.strapi.io).
:::

> ## Understand database migration files

Reviewer: I kind of like "Understanding" better personally 😆

Reviewer 2: I am trying to move away from -ing titles because they don't work well with translations, but I understand your preference. https://developers.google.com/style/headings

> ## Understand database migration files

Reviewer 3: ```suggestion
## Understanding database migration files
```

Migrations are run using JavaScript migration files stored in `./database/migrations`.

Strapi:

- Automatically detects migration files,

> - runs each new migration file 1 time in alphanumerical order at the next startup,

Reviewer: ```suggestion
- runs each new migration file one time in alphanumeric order at the next startup,
```

There is no CLI to manually execute the database migrations.
:::

## Create a migration file

Use the following procedure to create a migration file:

> 1. In the project root directory, run the command:

Reviewer 2: I don't understand the documentation as written. The file names start with the date, but we say the files run in alphabetical order. What does it really do, run in chronological order?

> 1. In the project root directory, run the command:

Reviewer: It runs in alphabetical order, which -should- be the same thing as chronological. But the code had changed the separator between the numbers, which would have caused them to run out of order chronologically. So the code has been updated to make sure the newly generated migrations match the existing documentation, which means no change needed in the docs.

There is no CLI to manually execute the database migrations.
:::

## Create a migration file

Use the following procedure to create a migration file:

1. In the project root directory, run the command:

    <code-group>
    <code-block title="YARN">

    ```bash
    yarn strapi generate migration
    ```

    </code-block>
    <code-block title="NPM">

    ```bash
    npm run strapi generate migration
    ```

    </code-block>
    </code-group>

> 2. Type a filename when prompted. The output filename is a concatenation of a date-time string and your filename with the format: `YYYY.MM.DDTHH.MM.SS.name-of-my-migration.js`

Reviewer: ```suggestion
2. Type a migration name when prompted. The output filename is a concatenation of a date-time string and your filename with the format: `YYYY.MM.DDTHH.MM.SS.name-of-my-migration.js`
```

Reviewer 4: Do we allow configuration of this format somewhere? I've never seen that date format used with periods.

Typically it should be `YYYY-MM-DD_HH:MM:SS_name-of-file.js`
